### DISCUSSION TITLE
What is the proper way to show different content for offcanvas show menu versus expanded menu list?

### DISCUSSION THREAD

**Discussion by tomasts248** [twbs/bootstrap#37576] [Q&A] [answered]
I know I could create two divs with d-none that toggle with a @media CSS rule based on the expanded class. But is that the correct way or there's a better one?

**Comment by tomasts248** [comment]
Ok I used breakpoint d-none classes stil wonder if there s a better way...

**Comment by gianlucajahn** [comment, marked as answer]
You can use the .collapse and .show classes to show different content for the offcanvas show menu and the expanded menu list.

To do this, you can add the .collapse class to the offcanvas menu element, and then add the .show class to the expanded menu list element. This will hide the offcanvas menu by default, and show the expanded menu list instead. Then, when the offcanvas menu is toggled, the .collapse and .show classes will be toggled on the corresponding elements, which will show the offcanvas menu and hide the expanded menu list.

```
<!-- Offcanvas menu, hidden by default -->
<nav class="collapse">
  <ul>
    <li><a href="#">Offcanvas menu item 1</a></li>
    <li><a href="#">Offcanvas menu item 2</a></li>
    <li><a href="#">Offcanvas menu item 3</a></li>
  </ul>
</nav>

<!-- Expanded menu list, shown by default -->
<nav class="show">
  <ul>
    <li><a href="#">Expanded menu item 1</a></li>
    <li><a href="#">Expanded menu item 2</a></li>
    <li><a href="#">Expanded menu item 3</a></li>
  </ul>
</nav>
```

In this example, we have added the .collapse class to the offcanvas menu element, and the .show class to the expanded menu list element. This will hide the offcanvas menu and show the expanded menu list by default. When the offcanvas menu is toggled, the .collapse and .show classes will be toggled on the corresponding elements, which will show the offcanvas menu and hide

**Reply by tomasts248** [reply]
Amazing thank you so much!!
